ecosystem/indexer-grpc/indexer-grpc-utils/src/storage_format.rs

```diff
@@ -0,0 +1,513 @@
+        }
+        if starting_version % FILE_ENTRY_TRANSACTION_COUNT != 0 {
+            panic!("Starting version has to be a multiple of FILE_ENTRY_TRANSACTION_COUNT.")
+        }
+        match storage_format {
+            StorageFormat::GzipCompressedProto => {
+                let t = TransactionsInStorage {
+                    starting_version: Some(transactions.first().unwrap().version),
+                    transactions,
+                };
+                t.encode(&mut bytes).expect("proto serialization failed.");
+                let mut compressed = GzEncoder::new(bytes.as_slice(), flate2::Compression::fast());
+                let mut result = Vec::new();
+                compressed
+                    .read_to_end(&mut result)
+                    .expect("Gzip compression failed.");
+                FileEntry::GzipCompressionProto(result)
+            },
+            StorageFormat::Base64UncompressedProto => {
+                panic!("Base64UncompressedProto is not supported.")
+            },
+            StorageFormat::JsonBase64UncompressedProto => {
+                let transactions_in_base64 = transactions
+                    .into_iter()
+                    .map(|transaction| {
+                        let mut bytes = Vec::new();
+                        transaction
+                            .encode(&mut bytes)
+                            .expect("proto serialization failed.");
+                        base64::encode(bytes)
+                    })
+                    .collect::<Vec<String>>();
+                let file = TransactionsLegacyFile {
+                    starting_version,
+                    transactions_in_base64,
+                };
+                let json = serde_json::to_vec(&file).expect("json serialization failed.");
+                FileEntry::JsonBase64UncompressedProto(json)
+            },
+        }
+    }
+
+    pub fn build_key(version: u64, storage_format: StorageFormat) -> String {
+        let starting_version =
+            version / FILE_ENTRY_TRANSACTION_COUNT * FILE_ENTRY_TRANSACTION_COUNT;
+        let mut hasher = Ripemd128::new();
+        hasher.update(starting_version.to_string());
+        let file_prefix = format!("{:x}", hasher.finalize());
+        match storage_format {
+            StorageFormat::GzipCompressedProto => {
+                format!("compressed_files/gzip/{}_{}.bin", file_prefix, version)
+            },
+            StorageFormat::JsonBase64UncompressedProto => {
+                format!("files/{}.json", starting_version)
+            },
+            StorageFormat::Base64UncompressedProto => {
+                panic!("Base64UncompressedProto is not supported.")
+            },
+        }
+    }
+}
+
+impl TryFrom<FileEntry> for TransactionsInStorage {
+    type Error = anyhow::Error;
+
+    fn try_from(value: FileEntry) -> Result<Self, Self::Error> {
```

Comment:
same thing here. could we move this into a function `FileEntry::to_transactions`

Reply:
done.